docs/modules/ROOT/pages/node_types.adoc

```diff
@@ -0,0 +1,234 @@
+= Node Types
+
+This is a partial list of the node types parsed by the AST and corresponding methods and information associated with them.
+
+`Rubocop::AST` defines some additional methods on certain node types by giving them a separate class. You can see the method definitions in the https://rubydoc.info/github/rubocop-hq/rubocop-ast/RuboCop/AST/Node[API documentation].
+
+For full information, please see the https://github.com/whitequark/parser/blob/master/doc/AST_FORMAT.md[parser documentation]. This page will act as a quick index to that page, which has more examples and fuller explanation of the different location pieces.
+
+There are a few "meta-types" of nodes that will be mentioned in descriptions:
+
+* Expression nodes: any expression that returns a value - variables (`lvar`, `ivar`, `cvar` etc.), `send` or `csend`, `const`, `self`, any literal value (`int`, `str`, `nil`, etc.), or control statements like `if`, `case`, `begin`, etc.
+* Assignment nodes: Any node that assigns a value. `lvasgn`, `ivasgn`, `cvasgn`, `gvasgn`, `send` (e.g. `self.foo = 5`), or `csend`.
+* Block statements: These can be essentially any node except those that must be nested (like `args` or `mlhs`). Typically they come as the children of a node that can contain arbitrary code, like `def` or `class`. If there are no statements, this will be a single `nil` node.
+
+== Location Information
+
+There are different parts of the source map you can get from the node, by calling `.loc` on it. Every node has an `expression` value which
+usually represents the entire node, but others have additional fields.
+
+The following fields are given when relevant to nodes in the source code:
+
+[cols="m,a"]
+|=========
+|Field|Description
+|assoc|The fat-arrow: `=>` used in `resbody` (NOT in hash `pair` s)
+|begin|* Start of parentheses or square brackets `(` or `[`
+* Start of `do..end` blocks (containing the `do` keyword)
+* `then` keyword
+* `begin` keyword
```

Comment:
If you want to be complete, it's also what surrounds literals like quotes for strings, the leading colon for symbols. BTW I found a [long standing bug in parser](https://github.com/whitequark/parser/pull/715) while reviewing your doc :-)